File: students/students/src/MON.SPA/src/models/studentResourceSupportModel.js
import Constants from '@/common/constants.js';
import { DocumentModel } from '@/models/documentModel.js';

export class StudentResourceSupportModel {
  constructor(obj = {}, moment = {}) {
    this.id = obj.id || null;
    this.personId = obj.personId || null,
    this.schoolYear = obj.schoolYear || null,
    this.reportNumber = obj.reportNumber || null,
    this.reportDate = (obj.reportDate && moment) ? moment(obj.reportDate).format(Constants.DATEPICKER_FORMAT) : '';
    this.documents = obj.documents ? obj.documents.map(el => new DocumentModel(el)) : [];
    this.resourceSupports = obj.resourceSupports || [];
  }
}

File: students/students/src/MON.Models/StudentModels/StudentResourceSupportModel.cs
using System;
using System.Collections.Generic;

namespace MON.Models.StudentModels
{
    public class StudentResourceSupportModel
    {
        public int? Id { get; set; }
        public int PersonId { get; set; }
        public short SchoolYear { get; set; }
        public string ReportNumber { get; set; }
        public DateTime ReportDate { get; set; }
        public IEnumerable<DocumentViewModel> Documents { get; set; }
        public List<ResourceSupportModel> ResourceSupports { get; set; } = new List<ResourceSupportModel>();
    }
}
